pkg/fees/calculator.go

```diff
@@ -0,0 +1,73 @@
+package fees
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/xmtp/xmtpd/pkg/currency"
+)
+
+type FeeCalculator struct {
+	ratesFetcher IRatesFetcher
+}
+
+func NewFeeCalculator(ratesFetcher IRatesFetcher) *FeeCalculator {
+	return &FeeCalculator{ratesFetcher: ratesFetcher}
+}
+
+func (c *FeeCalculator) CalculateBaseFee(
+	messageTime time.Time,
+	messageSize int64,
+	storageDurationDays int64,
```

Comment:
you can probably get away with `uint32` for both

Reply:
We can, but then we immediately have to convert to multiply against the int64 picodollars